src/vs/workbench/parts/extensions/browser/extensionsActions.ts

```diff
@@ -1133,6 +1134,120 @@ export class ShowWorkspaceRecommendedExtensionsAction extends Action {
 	}
 }
 
+export class InstallWorkspaceRecommendedExtensionsAction extends Action {
+
+	static ID = 'workbench.extensions.action.installWorkspaceRecommendedExtensions';
+	static LABEL = localize('installWorkspaceRecommendedExtensions', "Install All Workspace Recommended Extensions");
+
+	private disposables: IDisposable[] = [];
+	private recommendations: string[] = [];
+	private toInstall: string[] = [];
+	private updatePromise: TPromise<any> = TPromise.as(null);
+
+	constructor(
+		id: string = InstallWorkspaceRecommendedExtensionsAction.ID,
+		label: string = InstallWorkspaceRecommendedExtensionsAction.LABEL,
+		@IWorkspaceContextService private contextService: IWorkspaceContextService,
+		@IViewletService private viewletService: IViewletService,
+		@IExtensionsWorkbenchService private extensionsWorkbenchService: IExtensionsWorkbenchService,
+		@IExtensionTipsService private extensionTipsService: IExtensionTipsService,
+		@IMessageService private messageService: IMessageService
+	) {
+		super(id, label, 'extension-action');
+		this.extensionsWorkbenchService.onChange(() => this.updateToInstall(), this, this.disposables);
+		this.contextService.onDidChangeWorkbenchState(() => this.updateRecommendations(), this, this.disposables);
+		this.updatePromise = this.updateRecommendations();
+	}
+
+	private updateRecommendations(): TPromise<any> {
+		this.enabled = this.contextService.getWorkbenchState() !== WorkbenchState.EMPTY;
+		if (this.enabled) {
+			return this.extensionTipsService.getWorkspaceRecommendations().then(names => {
+				this.recommendations = names;
+				this.updateToInstall();
+			});
+		}
+		return TPromise.as(null);
+	}
+
+	private updateToInstall(): void {
+		const installed = this.extensionsWorkbenchService.local.map(x => x.id.toLowerCase());
+		this.toInstall = this.recommendations.filter(x => installed.indexOf(x.toLowerCase()) === -1);
+		this.enabled = this.toInstall.length > 0;
+	}
+
+	run(): TPromise<any> {
+		this.viewletService.openViewlet(VIEWLET_ID, true)
+			.then(viewlet => viewlet as IExtensionsViewlet)
+			.then(viewlet => {
+				viewlet.search('@recommended:workspace ');
+				viewlet.focus();
+			});
+
+		return this.updatePromise.then(() => {
+			if (!this.toInstall.length) {
+				this.messageService.show(Severity.Info, localize('allExtensionsInstalled', "All extensions recommended for this workspace have already been installed"));
+				return TPromise.as(null);
+			}
+			return this.extensionsWorkbenchService.queryGallery({ names: this.toInstall, source: 'install-all-workspace-recommendations' }).then(pager => {
+				return TPromise.join(pager.firstPage.map(e => this.extensionsWorkbenchService.install(e)));
```

Comment:
What happens if there are more than one page?

Reply:
One page holds about 50 extensions. It would be very rare if there are over 50 workspace recommendations.
I'll hold off on changing this and see if there is any user feedback with someone falling in this category

Reply:
What is missing in installing all extensions if there are more than 50? Is it unexpected? I think we should do all or none. We already show the count badge on the section.

Reply:
Done. https://github.com/Microsoft/vscode/commit/2d0b55460d380b8693b9b9151021d1671f15268f